driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/execve.bpf.c

```diff
@@ -99,10 +99,10 @@ int BPF_PROG(execve_x,
 		unsigned long argv = extract__syscall_argument(regs, 1);
 
 		/* Parameter 2: exe (type: PT_CHARBUF) */
-		auxmap__store_single_charbuf_param_from_array(auxmap, argv, 0, USER);
+		auxmap__store_execve_exe(auxmap, (char **)argv);
```

Comment:
@Andreagit97 , I wonder whether we should also have upper boundary check when reading args / envp from `mm`.
Reading on `execve(2)`:

>        On Linux prior to kernel 2.6.23, the memory used to store the environment and argument strings was limited to 32 pages (defined by the kernel
       constant MAX_ARG_PAGES).  On architectures with a 4-kB page size, this yields a maximum size of 128 kB.

Later versions, provide more larger values based on `RLIMIT_STACK`. However, to my understanding so far, the event buffer is limited to 64KB. Similar to other places (`iovec`, `sockaddr`), we could use `SAFE_ACCESS()` (and would make BPF verifier more happier ;-)

Wdyt?

Reply:
> Similar to other places (iovec, sockaddr), we could use SAFE_ACCESS() (and would make BPF verifier more happier ;-)

Today the verifier never complains because our scratch map is on `128 KB` and we leave always the last `64 KB` empty, in this way we are sure that even if the length to read is 64 KB (max possible length for 16 bit) we have enough space to write our data

Reply:
Hi @Andreagit97 ,

> ```c
> static __always_inline u16 auxmap__store_charbuf_param(struct auxiliary_map *auxmap, unsigned long charbuf_pointer, unsigned long len_to_read, enum read_memory mem)
> ```
> 
> WDYT? @hbrueckner

I would prefer to add the `len_to_read` parameter for those reasons:
- it becomes consistent with `auxmap__store_bytebuf_param()` from a prototype perspective
- it enables the caller to control the maximum to read (and makes the limit more visible instead of implicitly limiting to `MAX_PARAM_SIZE`).

... and thanks for the explanation on the 16bit (`u16`) parameter regarding BPF verifier.

Reply:
@hbrueckner sorry for the late answer the libs release took us a lot of time :( I've changed the signature of `auxmap__store_charbuf_param` as we said. Unfortunately, I had to change many files, but it should be quite easy to review them. I've added some macros to address the different use cases (this should also satisfy @incertum request for more control over the max size):
- `MAX_PROC_EXE`
- `MAX_PROC_ARG_ENV`
- `MAX_PATH`

They point all to the same value but I would prefer to keep them conceptually separated, maybe in the next future they could become also BPF `const` global variables chosen at load time :thinking:

Reply:
See other comment above - we may need an overall cleanup PR across drivers and userspace for all upper limits - as there appear to be inconsistencies. @Andreagit97  this is a great improvement, thanks for addressing this so quickly and we can iterate on it - no need to block this PR aka LGTM. In case you both @hbrueckner and @Andreagit97 have time to peak into more problems we may have for actual threat detection use cases, created https://github.com/falcosecurity/falco/issues/2259 - however with no expectations re if it's even feasible to improve without sacrificing perf and stability.

Reply:
yeah, I think we definitely need it! Thank you for pointing this out :)